internal/pkg/agent/application/actions/handlers/handler_action_policy_change_test.go

```diff
@@ -1103,10 +1115,10 @@ func TestPolicyChangeHandler_handlePolicyChange_LogLevelSet(t *testing.T) {
 }
 
 func nilLogLevelSet(t *testing.T) *mockhandlers.LogLevelSetter {
-	// nilLogLevel is a variable used to match nil policy log level being set
-	var nilLogLevel *logger.Level = nil
+	// default log level should be set when policy log level is nil
+	defaultLogLevel := logger.DefaultLogLevel
```

Comment:
If you set Info here, how do you clear a log level coming from policy to allow fallback onto the agent default level ? Having a nil log level value from the policy should be handled in the loglevel setter since that's the object that has the logic for proper priority and fallback, this will force the agent to always have a log level coming from policy even when there is non specified